packages/components/stamp/src/stamp.tsx

```diff
@@ -133,17 +117,55 @@ const getStampStyles = (props: StylesFunctionParams) => {
 const Stamp = (props: Props) => {
   const { themedValue } = useTheme();
   const overrideTextColor = themedValue(false, true);
+  const iconTone = props.tone ? tonesStylesMap[props.tone].iconTone : '';
+  console.log({ iconTone });
+  const Icon =
+    props.icon &&
+    cloneElement(props.icon, {
+      size: 'medium',
+      color: overrideTextColor ? iconTone : 'inherit',
+    });
```

Comment:
I would suggest to extract the logic around the icon color into a function as it relies on the `props.tone` and `overrideTextColor`. Remove it then from the `tonesStylesMap`.

```ts
const getIconColor = (props: Props, overrideTextColor: boolean) => {
  if (!overrideTextColor) {
    return 'inherit';
  }
  switch (props.tone): {
    case 'critical':
      return 'error';
    // ...
    default:
      return '';
  }
}
```

We should also think if we want to have a default `tone`, or validate that one is provided.

Reply:
I'm hesitant about this. I believe having the map of styles tied to the `tone` provided to the `Stamp` helps in grouping concerns since that prop is the one that controls the other styles.

I created [this PR](https://github.com/commercetools/ui-kit/pull/2393/) with the way I see this could work where I think it would be easy to follow how the styles of the component work.